src/ocm/components/clusterDetail/ClusterDetailStatusVarieties.tsx

```diff
@@ -21,20 +21,23 @@ type ClusterStatusVarieties = {
   fetchCredentials: () => void;
 };
 
-export const useClusterStatusVarieties = (cluster: Cluster): ClusterStatusVarieties => {
+export const useClusterStatusVarieties = (cluster?: Cluster): ClusterStatusVarieties => {
```

Comment:
Why is it that we can get here without a cluster? I think in that case , the hook can't do anything so IMO it should still be mandatory.

Then if you are in a place where there's no cluster, conditionally avoid rendering the component using the hook, etc.

Reply:
Because in `AddHosts` (where we call the `useClusterStatusVarieties`) we get the cluster from `AddHostsContext` which starts empty. I cannot prevent calling the hook if the cluster is `undefined`, which shouldn't happen but Typescript will think it can.

Reply:
If I made `cluster` mandatory in AddHostsContext, I'd have to define it [here](https://github.com/openshift-assisted/assisted-ui-lib/blob/7520c65f81c3d5ac4540eae437ef883551a09a07/src/common/components/AddHosts/AddHostsContext.tsx#L10). I'm not sure how I could do that.

I've tried to change it so that it only triggers if `cluster.id` or `cluster.status` change (since these are the only fieldfs we read in there.